modules/apps/social/social-bookmarks-taglib/src/main/resources/META-INF/resources/js/SocialBookmarksHandleItemClick.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import openSocialBookmark from './openSocialBookmark';
+
+export default function () {
+	window.socialBookmarks_handleItemClick = function (
```

Comment:
Can't we avoid setting this inside `window`?

Reply:
Not really without reimplementation 😞  because the call is generated here https://github.com/liferay-lima/liferay-portal/pull/1568/commits/503526c113cd6132dffb349223efcf921c50ae67#diff-b54940963e7c40c8ef25279c6df68bd1c97f92b62ec9081194ac887af4a073fdR51.

Reply:
I moved the code outside JSP but not change the logic at all

Reply:
You can probably do something like (pseudocode):

- Remove that inline onClick from the `li` element
- Pass the function name to the component

```
<liferay-frontend:component
    context="{socialBookmarksClickAction: <%= SocialBookmarksTagUtil.getClickJSCall(className, classPK, types[i], socialBookmark.getPostURL(title, url), url) %>}
    module="js/SocialBookmarksHandleItemClick"
/>
```
- Update the componente
```
export default function (socialBookmarksClickAction) {
	const socialBookmarksHandleItemClick = (
		event,
		className,
		classPK,
		type,
		postURL,
		url
	) => {
		event.preventDefault();
		event.stopPropagation();

		openSocialBookmark({className, classPK, postURL, type, url});
	};

	const ACTIONS = {
		socialBookmarks_handleItemClick: socialBookmarksHandleItemClick
	};

	document.querySelector('.taglib-social-bookmark').addEventListener(
		'click',
		ACTIONS[socialBookmarksClickAction]
	);
}
```

This is a non-blocking stuff since there's no logic change here, just curious about that use of window. If you find yourself in the mood to give it a try maybe we can open a thread in Slack to talk with FI guys about if storing methods in `window` is something acceptable or something we want to get rid of.

Reply:
I'm not sure if this idea works because socialBookmarksClickAction isn't the name of the function is the call itself for example `socialBookmarks_handleItemClick(event, 'com\x2eliferay\x2eblogs\x2emodel\x2eBlogsEntry', 39993, 'facebook', 'http\x3a\x2f\x2fwww\x2efacebook\x2ecom\x2fsharer\x2ephp\x3fu\x3dhttp\x253A\x252F\x252Flocalhost\x253A8080\x252Fhome\x252F-\x252Fblogs\x252Fasdasda', 'http\x3a\x2f\x2flocalhost\x3a8080\x2fhome\x2f-\x2fblogs\x2fasdasda');`

Reply:
Ok, then the refactor should be a bit more complex. Let's keep this as it is. Thx!